semaphore/semaphore.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2017 The Go Authors. All rights reserved.
```

Comment:
Can this directory also gain a README summarizing: what upstream sha1 / version number this was forked from, why it was forked, and what general changes have been made since then? I tried to do this for crl_x509: https://github.com/letsencrypt/boulder/blob/main/crl/crl_x509/README.md